app/gui2/src/components/GraphEditor/widgets/WidgetPort.vue

```diff
@@ -109,28 +110,32 @@ const enabled = computed(() => {
   return !isConditional || keyboard.mod
 })
 
-const computedRect = computed(() => {
+function relativePortSceneRect(): Rect | undefined {
   const domNode = rootNode.value
-  const rootDomNode = domNode?.closest('.GraphNode')
+  const rootDomNode = tree.nodeElement
   if (domNode == null || rootDomNode == null) return
   if (!enabled.value) return
-  let _nodeSizeEffect = nodeSize.value
   const exprClientRect = Rect.FromDomRect(domNode.getBoundingClientRect())
   const nodeClientRect = Rect.FromDomRect(rootDomNode.getBoundingClientRect())
   const exprSceneRect = navigator.clientToSceneRect(exprClientRect)
   const exprNodeRect = navigator.clientToSceneRect(nodeClientRect)
   return exprSceneRect.offsetBy(exprNodeRect.pos.inverse())
-})
+}
 
 function updateRect() {
-  const newRect = computedRect.value
-  if (!Rect.Equal(portRect.value, newRect)) {
+  const oldRect = portRect.value
+  const newRect = relativePortSceneRect()
+  if (
+    oldRect !== newRect &&
+    (oldRect == null || newRect == null || !oldRect.equalsApproximately(newRect, 0.01))
+  ) {
     portRect.value = newRect
   }
 }
 
-watch(computedRect, updateRect)
+watch(() => [nodeSize.value, rootNode.value, tree.nodeElement, tree.nodeSize], updateRect)
```

Comment:
- `relativePortSceneRect` could have a comment that its reactive dependencies are enforced externally and this needs to be kept in sync.
- How are conditional ports working when `enabled` is not watched?

Reply:
The `enabled` computed is used in the template, so the `onUpdated` hook was covering it. But that is quite opaque, so I just added it to the watched deps. Added a comment on top of `updateRect`, as this is really the entry point of this non-really-reactive effect.